tests/lib/devicetree/api/src/main.c

```diff
@@ -3326,6 +3326,25 @@ ZTEST(devicetree_api, test_string_idx_unquoted)
 #undef XMINUS
 }
 
+#undef DT_DRV_COMPAT
+ZTEST(devicetree_api, test_string_escape)
+{
+	zassert_str_equal(DT_PROP(DT_NODELABEL(test_str_escape_0), val), "\a\b\f\n\r\t\v");
+	zassert_str_equal(DT_PROP(DT_NODELABEL(test_str_escape_1), val), "\'single\' \"double\"");
+	zassert_str_equal(DT_PROP(DT_NODELABEL(test_str_escape_2), val), "first\nsecond");
+	zassert_str_equal(DT_PROP(DT_NODELABEL(test_str_escape_3), val), "\x01\x7F");
+}
+
+ZTEST(devicetree_api, test_string_array_escape)
+{
+	zassert_str_equal(DT_PROP_BY_IDX(DT_NODELABEL(test_stra_escape), val, 0),
+			  "\a\b\f\n\r\t\v");
+	zassert_str_equal(DT_PROP_BY_IDX(DT_NODELABEL(test_stra_escape), val, 1),
+			  "\'single\' \"double\"");
+	zassert_str_equal(DT_PROP_BY_IDX(DT_NODELABEL(test_stra_escape), val, 2), "first\nsecond");
+	zassert_str_equal(DT_PROP_BY_IDX(DT_NODELABEL(test_stra_escape), val, 3), "\x01\x7F");
+}
+
```

Comment:
why no testing of DT_STRING_UNQUOTED?

Reply:
Simply because I couldn't think of any reason you would ever want to use DT_STRING_UNQUOTED with these characters., since from the documentation it seems to be primarily for turning strings into C code, which typically would not use these characters.

Something like
```dts
val = "\"first\" SECOND \"third\"";
```
```c
#define SECOND " second "

zassert_str_equal(DT_STRING_UNQUOTED(node, val), "first second third")
```
could be a possible use for this though, so I will add some tests for it.

Reply:
yes, I think since you're going through the effort to make sure the generation of this macro handles the escaped strings, might as well add a test so that we are at least clear on the behavior for the future so that we can have some definition of what "handling" means